syncstorage/src/db/mod.rs

```diff
@@ -69,13 +69,24 @@ pub fn spawn_pool_periodic_reporter<T: GetPoolState + Send + 'static>(
     Ok(())
 }
 
-pub async fn run_on_blocking_threadpool<F, T>(f: F) -> Result<T, DbError>
+/// Runs a function as a task on Actix's blocking threadpool.
+///
+/// WARNING: Calling `web::block` anywhere else will result in inaccurate threadpool metrics being
+/// reported. If you want to spawn a task on Actix's blocking threadpool, you **must** use this
+/// function.
+pub async fn run_on_blocking_threadpool<F, T>(metrics: Metrics, f: F) -> Result<T, DbError>
 where
     F: FnOnce() -> Result<T, DbError> + Send + 'static,
     T: Send + 'static,
 {
-    web::block(f).await.map_err(|e| match e {
+    metrics.incr("blocking_tasks");
```

Comment:
I don't think there's a way to get this statistics from the Actix server/tokio runtime directly, so this was my best attempt to keeping track of these tasks. The main downside to this (as reported above) is that if someone calls `web::block` directly somewhere else in the code base, the task spawned there would not be reported in the metrics